hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -395,18 +391,19 @@ protected ClientConnection createSocketConnection(final Address address) throws
             HazelcastProperties properties = client.getProperties();
             boolean directBuffer = properties.getBoolean(SOCKET_CLIENT_BUFFER_DIRECT);
 
-            Channel channel =
-                    channelFactory.create(socketChannel, true, directBuffer);
+            Channel channel = channelFactory.create(socketChannel, true, directBuffer);
 
             final ClientConnection clientConnection = new ClientConnection(
-                    client, eventLoopGroup, connectionIdGen.incrementAndGet(), channel);
+                    client, connectionIdGen.incrementAndGet(), channel);
             socketChannel.configureBlocking(true);
             if (socketInterceptor != null) {
                 socketInterceptor.onConnect(socket);
             }
-            socketChannel.configureBlocking(eventLoopGroup.isBlocking());
```

Comment:
The eventloopgroup takes care of configuring the socketchannel when the channel is registered.